realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -696,7 +696,8 @@ public void combiningAsyncAndSync() {
         final RealmResults<AllTypes> allTypesAsync = looperThread.realm.where(AllTypes.class).greaterThan("columnLong", 5).findAllAsync();
         final RealmResults<AllTypes> allTypesSync = allTypesAsync.where().greaterThan("columnLong", 3).findAll();
 
-        assertEquals(0, allTypesAsync.size());
+        // Call where() on an async results will load the async query immediately.
```

Comment:
nice, this will fix https://github.com/realm/realm-java/issues/3668#issuecomment-255378795